index.js

```diff
@@ -18,110 +18,170 @@ function parseDotenvFile(path, verbose = false) {
   return dotenv.parse(content)
 }
 
-module.exports = ({types: t}) => ({
-  name: 'dotenv-import',
-
-  pre() {
-    this.opts = {
-      envName: 'APP_ENV',
-      moduleName: '@env',
-      path: '.env',
-      whitelist: null,
-      blacklist: null,
-      allowlist: null,
-      blocklist: null,
-      safe: false,
-      allowUndefined: true,
-      verbose: false,
-      ...this.opts,
+module.exports = (api, options) => {
+  const t = api.types
+  options = {
+    envName: 'APP_ENV',
+    moduleName: '@env',
+    path: '.env',
+    whitelist: null,
+    blacklist: null,
+    allowlist: null,
+    blocklist: null,
+    safe: false,
+    allowUndefined: true,
+    verbose: false,
+    ...options,
+  }
+  const babelMode = process.env[options.envName] || (process.env.BABEL_ENV && process.env.BABEL_ENV !== 'undefined' && process.env.BABEL_ENV !== 'development' && process.env.BABEL_ENV) || process.env.NODE_ENV || 'development'
```

Comment:
You might want to wrap this in `cache.using`, so that if the env changes without restarting Node.js the plugin is reinstantiated.

Reply:
thank you! I moved things around. I still don't know if it will reload properly in hot reload, but I don't think the environment mode can change without a js reload so I left that out. Hopefully it works in other tests when we release it in `@next`!

Reply:
I am the second maintainer to this project so I'm not even sure about how all the babel features work